pkg/metrics/server_test.go

```diff
@@ -110,3 +129,102 @@ func TestMetricsServerPanic(t *testing.T) {
 		})
 	}
 }
+
+// getFreePort asks the kernel for a free open port that is ready to use.
+func getFreePort() (port int, err error) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		return 0, err
+	}
+	listener, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return 0, err
+	}
```

Comment:
can be minimized to:
```suggestion
	listener, err := net.Listen("tcp", "localhost:0")
	if err != nil {
		return 0, err
	}
```

But, since you only want to get the port, you should skip the whole listener part and simply do:
```go
	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
	if err != nil {
		return 0, err
	}
        return addr.Port, nil
```

Reply:
The first snippet works, but the second one does not - it will just return `127.0.0.1:0` instead of the actual resolved port. But we need the concrete port to send requests in the tests.